my_enumerable.rb

```diff
@@ -0,0 +1,210 @@
+      else
+        my_each do |value|
+          break result = true if yield(value) == true  end
+      end
+    end
+    result
+  end
+
+  def my_all?(args = nil)
+    result = true
+    unless args.nil?
+      if is_a? Hash
+        my_each do |_item, value|
+        break result = false unless value.class == args end
+      elsif args.is_a? Regexp
+        my_each do |value| 
+          result = false  if (value =~ args).nil?        
+        end
+      elsif (args == String) || (args == Integer) ||  (args == Float) ||  (args == Numeric)
+        my_each do |value| 
+          break result = false unless value.class == args  
+        end
+      else
+        my_each do |value|
+        break result = false unless value == args end
+      end
+    end
+    unless block_given? && args.nil?
+      if is_a? Hash
+        my_each do |item|
+        result = false if item[1] == false || item[1].nil? end
+      else
+        my_each do |item|
+        result = false if item == false || item.nil? end
+      end
+    else
+      if is_a? Hash
+        my_each do |item, value|
+        break result = false if yield(item, value) == false || yield(item, value).nil? end
+      else
+        my_each do |value|
+          break result = false if yield(value) == false || yield(value).nil?
+        end
+      end
+    end
+    result
+  end
+
+  def my_none?(args = nil)
+    result = true
+    unless args.nil?
+     if is_a? Hash
+        my_each do |_item, value|
+        break result = false unless value.class == args end
+     elsif args.is_a? Regexp
+        my_each do |value| 
+          break result = false  unless (value =~ args).nil?        
+        end
+     elsif (args == String) || (args == Integer) ||  (args == Float) ||  (args == Numeric)
+        my_each do |value| 
+          break result = false if value.class == args  
+        end
+      else
+        my_each do |value|
+        break result = false if value == args end
+      end
+    end
+    unless block_given? && args.nil?
+      if is_a? Hash
+        my_each do |item|
+        result = false if item[1] == true end
+      else
+        my_each do |item|
+        result = false if item == true end
+      end
+    else
+      if is_a? Hash
+        my_each do |item, value|
+        break result = false if yield(item, value) == true end
+      else
+        my_each do |value|
+          break result = false if yield(value) == true
+        end
+      end
+    end
+    result
+  end
+
+  def my_count(args = nil)
+    result = size
+    count = 0
+    if !args.nil?
+      my_each do |item|
+      item == args ? count += 1 : nil end
+      result = count
+    elsif block_given?
+      my_each do |item|
+      count += 1 if yield(item) == true end
+      result = count
+    end
+    result
+  end
+
+  def my_map
+    result = []
+    if block_given? 
+      my_each do  |item| 
+        result.push(yield(item)) end
+      return result
+    else
+      my_each
+    end
+   
+  end
```

Comment:
- [ ] my_map

returns an Enumerator if no block is given(Just like the original method):
`p [1, 2, 3].my_map` returns `#<Enumerator>`